src/protocol/boolean/bitwise_sum.rs

```diff
@@ -0,0 +1,212 @@
+        let ctx = make_contexts::<F>(&world);
+        let mut rand = StepRng::new(1, 1);
+
+        let a_bits = shared_bits(a, F::Integer::BITS, &mut rand);
+        let b_bits = shared_bits(b, F::Integer::BITS, &mut rand);
+        let l = a_bits.len();
+
+        let step = "BitwiseSum_Test";
+        let result = try_join_all(vec![
+            BitwiseSum::execute(
+                ctx[0].narrow(step),
+                RecordId::from(0_u32),
+                &transpose(&a_bits, 0),
+                &transpose(&b_bits, 0),
+            ),
+            BitwiseSum::execute(
+                ctx[1].narrow(step),
+                RecordId::from(0_u32),
+                &transpose(&a_bits, 1),
+                &transpose(&b_bits, 1),
+            ),
+            BitwiseSum::execute(
+                ctx[2].narrow(step),
+                RecordId::from(0_u32),
+                &transpose(&a_bits, 2),
+                &transpose(&b_bits, 2),
+            ),
+        ])
+        .await
+        .unwrap();
+
+        // `result` is comprised of three bitwise-sharings of `a + b`
+        let sum = (0..result[0].len())
+            .map(|i| validate_and_reconstruct((result[0][i], result[1][i], result[2][i])))
+            .collect::<Vec<_>>();
+
+        // Output's bit length should be `input.len() + 1`
+        assert_eq!(l + 1, sum.len());
+
+        Ok(sum)
+    }
+
+    #[tokio::test]
+    pub async fn fp31() -> Result<(), BoxError> {
+        let c = Fp31::from;
+
+        assert_eq!(c(0_u8), bits_to_field(&bitwise_sum(c(0), c(0)).await?));
+        assert_eq!(c(1), bits_to_field(&bitwise_sum(c(0), c(1)).await?));
+        assert_eq!(c(1), bits_to_field(&bitwise_sum(c(1), c(0)).await?));
+        assert_eq!(c(2), bits_to_field(&bitwise_sum(c(1), c(1)).await?));
+        assert_eq!(c(3), bits_to_field(&bitwise_sum(c(2), c(1)).await?));
+        assert_eq!(c(4), bits_to_field(&bitwise_sum(c(2), c(2)).await?));
+        assert_eq!(c(30), bits_to_field(&bitwise_sum(c(15), c(15)).await?));
+        assert_eq!(c(0), bits_to_field(&bitwise_sum(c(30), c(1)).await?));
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    pub async fn fp_32bit_prime() -> Result<(), BoxError> {
+        let c = Fp32BitPrime::from;
+
+        assert_eq!(c(0_u32), bits_to_field(&bitwise_sum(c(0), c(0)).await?));
+        assert_eq!(c(1), bits_to_field(&bitwise_sum(c(0), c(1)).await?));
+        assert_eq!(c(1), bits_to_field(&bitwise_sum(c(1), c(0)).await?));
+        assert_eq!(c(2), bits_to_field(&bitwise_sum(c(1), c(1)).await?));
+        assert_eq!(
+            c(2_147_483_648_u32),
+            bits_to_field(&bitwise_sum(c(2_147_483_647), c(1)).await?)
+        );
+        assert_eq!(
+            c(4_294_967_290),
+            bits_to_field(&bitwise_sum(c(2_147_483_645), c(2_147_483_645)).await?)
+        );
+        assert_eq!(
+            c(0),
+            bits_to_field(&bitwise_sum(c(2_147_483_645), c(2_147_483_646)).await?)
+        );
```

Comment:
This one seems like a place where rather than testing specific, hard-coded examples, we could easily just generate two random numbers, compute what the sum should be, and validate that it all worked correctly. That would get us test coverage of many more corner-cases.